examples/todolist/uniffi.toml

```diff
@@ -0,0 +1,5 @@
+[bindings.geckojs.threading]
+-default = "worker"
+set_default_list = "direct"
+TodoList.-constructor = "direct"
+TodoList.make_default = "direct"
```

Comment:
Maybe "main" makes things more clear than "direct".

Reply:
I like that this is in the toml vs udl as you mentioned it keeps the udl clear from any platform-specific markup. 

I'm not too much of a fan of the dashes as it doesn't give a clear indication of why we need the dashes (if someone brand new is looking at this)

This is just my ideas but don't feel super strongly about it, but some bike-shedding ideas: 
- instead of `default` maybe `default_threading_mode` or just `default_thread_mode`?
- I like `main` as this essentially comes down to worker thread vs main thread

Example with suggestions above:
```toml
[bindings.geckojs.threading]
default_threading_mode = "worker"
set_default_list = "main"
TodoList.-constructor = "main"
TodoList.make_default =  "main"
```

A slightly crazier idea could be to just have a list denoting either main thread or worker thread (while obviously only needing whichever is not the default).
```toml
[bindings.geckojs.threading]
default_thread_mode = "worker"

[bindings.geckojs.threading.mainthread]
list_of_calls = [
  "set_default_list",
  "TodoList.constructor", 
  "TodoList.make_default"
]

Obviously these are just ideas so no worries if it's not in the spirit of what we're trying to do here!

Reply:
I agree this should be in the toml as it has no meaning on the rust side. OTOH though, the ergonomics of this is probably going to suck a little, so there might be a case to have binding-specific attributes on the UDL and/or in macro-based rust bindings, but this makes perfect sense to me to get the ball rolling.

If we can bash toml into playing nice, Sammy's idea of inverting things almost makes sense. If we consider that methods might end up with other attributes in the future, an alternative might also be to end up with something like:

```
...
[binding.geckojs.something.TodoList.set_default_list]
threading = "worker"
exception_behaviour = "something"
```

might also make sense (but might also make the ergonomic problem worse)

Finally, "threading" doesn't quite seem correct because it carries alot of baggage and comes with alot of assumptions. Something like "receiver_thread" or something *might* be more explicit, but I haven't thought that through, especially in the case of say, callbacks.

I suspect we will want a fixture in gecko to exercise some of this, but as mentioned, I think this is perfectly fine for a bit of a prototype and we can see how it shakes out in practice.